### PR TITLE
GitHub Pages Deploy

### PR DESCRIPTION
**Pull request by dfuchss**
Thanks for the nice admin ui.
I just thought about deploying the latest version to GitHub Pages since that makes the application easier to test/use by admins.
Therefore, you might think about merging this PR.

You can see the actual result on the [GH Page of my Fork](https://dfuchss.github.io/synapse-admin)

### CODE REVIEW

File: .github/workflows/edge_ghpage.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Build and Deploy Edge version to GH Pages
+on:
+ workflow_dispatch:
+ push:
```
Comment (dklimpel) [inline .github/workflows/edge_ghpage.yml:4]: Your file does not pass `yarn prettier`.
These changes are result of `yarn fix`.

```suggestion
  workflow_dispatch:
  push:
```

File: .github/workflows/edge_ghpage.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    branches:
+      - main
+      - master
+
```
Comment (dklimpel) [inline .github/workflows/edge_ghpage.yml:8]:
```suggestion

```

File: .github/workflows/edge_ghpage.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      - main
+      - master
+      
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2.3.1
+
```
Comment (dklimpel) [inline .github/workflows/edge_ghpage.yml:15]:
```suggestion

```

File: .github/workflows/edge_ghpage.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2.3.1
+      
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14"
+
```
Comment (dklimpel) [inline .github/workflows/edge_ghpage.yml:19]:
```suggestion

```

**Review by dklimpel — CHANGES_REQUESTED** [review]

**Comment by dfuchss** [comment]
I've accepted your change requests :)

**Comment by dfuchss** [comment]
Since this PR is now open approx. 4 months, I would like the maintainers what they think about the PR.

For me it would be the best to either close the PR or to merge it. I can't have a look at old open PRs and the dependencies of it might change and might need some update in the future.

**Comment by awesome-michael** [comment]
Thanks for your contribution. You're right and I'm sorry that it took so long. We had some internal discussions and lost focus a bit.. 
Will merge it now.

**Comment by dfuchss** [comment]
@awesome-michael thanks a lot for the response :)

I assume that you have to allow pushes via GH_TOKEN in the Actions Settings of the Repository.
image

**Comment by awesome-michael** [comment]
Thank you so much for looking into it.. I had no time to investigate further. Write permissions are greyed out at the moment, I'm searching right now..

**Comment by dfuchss** [comment]
You can also look into the actions settings of your organization. 

Otherwise, you could use a deployment ssh key https://github.com/JamesIves/github-pages-deploy-action#using-an-ssh-deploy-key-

or create a Personal Access Token for a "Bot" User.

**Comment by dfuchss** [comment]
@awesome-michael did you find the time to look at the problem in your permission settings?

**Comment by dklimpel** [comment]
It seems to work now.

**Comment by dfuchss** [comment]
👍 nice work , everyone can see the deployment in the list of environments :)

**Comment by awesome-michael** [comment]
Think I will add the link to the readme later..
Thanks for your work!

**Pull request merged**